src/sagemaker/tensorflow/deploying_python.rst

```diff
@@ -0,0 +1,190 @@
+to provide inference. ``predict`` will serialize the input data, and send it in as request to the ``SageMaker Endpoint`` by
+an ``InvokeEndpoint`` SageMaker operation. ``InvokeEndpoint`` operation requests can be made by ``predictor.predict``, by
+boto3 ``SageMaker.runtime`` client or by AWS CLI.
+
+The ``SageMaker Endpoint`` web server will process the request, make an inference using the deployed model, and return a response.
+The ``result`` returned by ``predict`` is
+a Python dictionary with the model prediction. In the code example above, the prediction ``result`` looks like this:
+
+.. code:: python
+
+  {'result':
+    {'classifications': [
+      {'classes': [
+        {'label': '0', 'score': 0.0012890376383438706},
+        {'label': '1', 'score': 0.9814321994781494},
+        {'label': '2', 'score': 0.017278732731938362}
+      ]}
+    ]}
+  }
+
+Specifying the output of a prediction request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The format of the prediction ``result`` is determined by the parameter ``export_outputs`` of the `tf.estimator.EstimatorSpec <https://www.tensorflow.org/api_docs/python/tf/estimator/EstimatorSpec>`_ that you returned when you created your ``model_fn``, see
+`Example of a complete model_fn`_ for an example of ``export_outputs``.
+
+More information on how to create ``export_outputs`` can find in `specifying the outputs of a custom model <https://github.com/tensorflow/tensorflow/blob/r1.4/tensorflow/docs_src/programmers_guide/saved_model.md#specifying-the-outputs-of-a-custom-model>`_.
+
+Endpoint prediction request handling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Whenever a prediction request is made to a SageMaker Endpoint via a ``InvokeEndpoint`` SageMaker operation, the request will
+be deserialized by the web server, sent to TensorFlow Serving, and serialized back to the client as response.
+
+The TensorFlow Web server breaks request handling into three steps:
+
+-  input processing,
+-  TensorFlow Serving prediction, and
+-  output processing.
+
+The SageMaker Endpoint provides default input and output processing, which support by default JSON, CSV, and protobuf requests.
+This process looks like this:
+
+.. code:: python
+
+    # Deserialize the Invoke request body into an object we can perform prediction on
+    deserialized_input = input_fn(serialized_input, request_content_type)
+
+    # Perform prediction on the deserialized object, with the loaded model
+    prediction_result = make_tensorflow_serving_prediction(deserialized_input)
+
+    # Serialize the prediction result into the desired response content type
+    serialized_output = output_fn(prediction_result, accepts)
+
+The common functionality can be extended by the addiction of the following two functions to your training script:
+
+Overriding input preprocessing with an ``input_fn``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
extend the header lines on ll. 153 and 172

Reply:
ok